### PR TITLE
feat: integrate axeReporterEarl results with the testrunner

### PR DESCRIPTION
**Pull request by jeeyyy**
This PR does the below:

- integrate `axeReporterEarl`.
- clean-up.
- generated EARL report and uploaded to - https://github.com/w3c/earl/blob/e439752f5a58eb5eb70d7ef0e5973282babf7606/earl-reports/axe-report.json

### CODE REVIEW

File: src/evaluate.js [hunk 1]
```diff
@@ -1,38 +1,33 @@
...
+  return new Promise(async (resolve, reject) => {
     const axeIds = rulesMap[testcase.ruleId]
     axe.run(
-      document,
       {
+        reporter: async function(raw, _, cb) {
+          const url = window.location.href
+          const results = await reporter(raw, url)
+          return results
+        },
```
Comment (WilcoFiers) [inline src/evaluate.js:13]: Reporter isn't an async function. How does this work?

File: src/evaluate.js [hunk 2]
```diff
@@ -1,38 +1,33 @@
...
         runOnly: {
           type: 'rules',
           values: axeIds
         }
       },
-      (err, result) => {
+      async (err, result) => {
```
Comment (WilcoFiers) [inline src/evaluate.js]: This isn't an async function either.

File: src/index.js [hunk 3]
```diff
@@ -2,17 +2,19 @@ const testRunner = require('testrunner')
...
       outputDirectory: 'output',
-      results
+      fileContent: results
     })
-  })
-  .catch(error => {
-    throw new Error(error)
-  })
+  } catch (err) {
+    throw new Error(err)
```
Comment (WilcoFiers) [inline src/index.js]: You don't need the try / catch statement here if you're just going to catch and re-throw the error.

File: src/utils.js [hunk 4]
```diff
@@ -1,173 +1,4 @@
 const fsExtra = require('fs-extra')
-const markdownTable = require('markdown-table')
```
Comment (WilcoFiers) [inline src/utils.js:2]: Why did you remove all of this stuff? Could still be useful to have our own reports some place, no?

**Review by WilcoFiers — DISMISSED** [review]

File: src/utils.js (hunk 4 above)
Comment (jeeyyy) [reply src/utils.js:2]: Don't think we need to generate `markdown-table`. It is less useful now that validations are done by ACC.

File: src/configuration.js [hunk 5]
```diff
@@ -1,9 +1,9 @@
+// const axePath = require.resolve('axe-core')
```
Comment (stephenmathieson) [inline src/configuration.js]: What is this for?

File: output/act-axe-result.json [hunk 6]
```diff
@@ -1,870 +1,8665 @@
...
-      {
-        "ruleId": "SC1-1-1-image-has-name",
-        "testcaseUrl": "https://auto-wcag.github.io/auto-wcag/auto-wcag-testcases/assets/SC1-1-1-image-has-name_inapplicable_example_3.html",
-        "testcaseStatus": {}
-      },
-      {
-        "ruleId": "SC1-1-1-image-has-name",
-        "testcaseUrl": "https://auto-wcag.github.io/auto-wcag/auto-wcag-testcases/assets/SC1-1-1-image-has-name_inapplicable_example_4.html",
-        "testcaseStatus": {}
+    "@context": {
```
Comment (stephenmathieson) [inline output/act-axe-result.json:3]: Do we need this 9,400 LOC file checked into Git?

File: package.json [hunk 7]
```diff
@@ -18,21 +19,21 @@
...
   },
   "homepage": "https://github.com/dequelabs/axe-act-testrunner#readme",
   "dependencies": {
-    "axe-core": "^3.1.2",
+    "axe-core": "^3.1.2-canary.e67fc65",
+    "axe-reporter-earl": "0.1.0-canary.be38c0de",
     "fs-extra": "^7.0.0",
-    "markdown-table": "^1.1.2",
-    "testrunner": "git+https://github.com/auto-wcag/testrunner.git"
+    "testrunner": "git+https://github.com/auto-wcag/testrunner.git#master"
```
Comment (stephenmathieson) [inline package.json:32]: It'd be nice to not rely on GitHub here. Is this package not in `npm`?

**Review by stephenmathieson — DISMISSED** [review]
Few small things.

File: output/act-axe-result.json (hunk 6 above)
Comment (jeeyyy) [reply output/act-axe-result.json:3]: Sadly yes. We need this, as it is referenced from other projects like in `auto-wcag`

File: package.json (hunk 7 above)
Comment (jeeyyy) [reply package.json:32]: Yup, would be nice. But this is not pushed to npm and no plans just yet.
Once we release that, I will be updating here, but will have to be a different PR.

**Review by WilcoFiers — APPROVED** [review]
